vyper/builtin_functions/functions.py

```diff
@@ -1711,7 +1712,56 @@ def build_LLL(self, expr, args, kwargs, context):
         return LLLnode(None, typ=output_type, pos=getpos(expr))
 
 
+class ABIEncode(_SimpleBuiltinFunction):
+    _id = "abi_encode"  # TODO prettier to rename this to abi.encode
```

Comment:
If this function were namespaced to `ethereum.abi` that would be awesome e.g.

```python
from ethereum.abi import encode as abi_encode
...
_ = abi_encode(...)
# OR
from ethereum import abi
...
_ = abi.encode(...)
```

Reply:
I'm not sure if that's in scope with this PR, I think that might introduce some ad-hoc rules (that are hard to refactor later) into the name resolution code. I guess I could do it but instead I was thinking we should release this function as _abi_encode (to indicate to users it is kind of in beta) and then open a discussion VIP for the namespacing/naming piece. What do you think?

Reply:
We can do that. Namespacing as a general thing is indeed a bigger change, but something we could easily do for v0.3.0 to reduce the burden of checking for reserved keywards